### PR TITLE
stop propagation of clicks on tooltips

### PR DESCRIPTION
**Pull request by snow893**
Closes https://github.com/adobe/react-spectrum/issues/4480

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Open new storybook entry. Click red box. Button toggles disabled state. Try clicking on tooltip, click event is not propagated to the red div.

### CODE REVIEW

File: packages/@react-spectrum/tooltip/test/Tooltip.test.js [hunk 1]
```diff
@@ -40,4 +40,15 @@ describe('Tooltip', function () {
...
   });
+
+  it('click does not propagate to parent', () => {
+    let mockClick = jest.fn();
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions
+    let {getByRole} = render(<div onClick={mockClick}><Tooltip>This is a tooltip</Tooltip></div>);
+    let tooltip = getByRole('tooltip');
+    fireEvent.pointerDown(tooltip);
+    fireEvent.click(tooltip);
+    fireEvent.mouseDown(tooltip);
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/test/Tooltip.test.js:51]: Technically this is the wrong order of events compared to the browser. Lets simplify and use our util function to handle this.
```suggestion
    triggerPress(tooltip);
```

File: packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx [hunk 2]
```diff
@@ -176,6 +176,24 @@ export const TooltripTriggerInsideActionGroup: TooltipTriggerStory = {
...
+      <TooltipTrigger {...args} isOpen>
+        <ActionButton isDisabled={isDisabled} aria-label="Edit" UNSAFE_style={{top: '50px'}}>click red to disable</ActionButton>
+        <Tooltip>Click on tooltip doesn't propagate to parent</Tooltip>
+      </TooltipTrigger>
+    </div>
+  );
+};
+export const TooltripTriggerInsideDiv: TooltipTriggerStory = {
+  args: {delay: 0},
+  render: TooltipDivRender
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx:194]:
```suggestion
  render: TooltipDivRender,
  parameters: {description: {data: 'Event handlers are attached to the parent of the tooltip trigger. They should not be called when the tooltip itself is clicked.'}}
```

File: packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx [hunk 3]
```diff
@@ -176,6 +176,24 @@ export const TooltripTriggerInsideActionGroup: TooltipTriggerStory = {
   )
 };
 
+const TooltipDivRender = (args) => {
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx:179]: bit weird, this is a render function, not a component, could cause problems if storybook changes how their internals work

instead, make this a component taking props (looks exactly the same, just change args to props)
and then down below for render you should call `(args) => <TooltipDivRender {...args} />`

**Review by snowystinger — COMMENTED** [review]
Great, looks like it's working.

Would you mind adding a test to TooltipTrigger tests as well?

```
  it('does not propagate pointer or click through the portal', () => {
    let onPointerDown = jest.fn();
    let onPointerUp = jest.fn();
    let onClick = jest.fn();

    let {getByRole} = render(
      <Provider theme={theme}>
        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
        <div onPointerDown={onPointerDown} onPointerUp={onPointerUp} onClick={onClick}>
          <TooltipTrigger>
            <ActionButton>Trigger</ActionButton>
            <Tooltip>Helpful information.</Tooltip>
          </TooltipTrigger>
        </div>
      </Provider>
    );

    let button = getByRole('button');
    act(() => {
      button.focus();
    });

    let tooltip = getByRole('tooltip');
    triggerPress(tooltip);
    expect(onPointerDown).not.toHaveBeenCalled();
    expect(onPointerUp).not.toHaveBeenCalled();
    expect(onClick).not.toHaveBeenCalled();
  });
```

File: packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx (hunk 3 above)
Comment (snow893) [reply packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx:179]: makes sense. yeah i'm still getting used to this style of stories, wasn't getting it to work how i'd expected

**Review by snowystinger — APPROVED** [review]

**Review by LFDanLu — COMMENTED** [review]
A bit concerned about using usePress here just to stop propagation of click events, especially since usePress handles so much extra stuff like preventing text selection. Any reason we don't just use a pointerdown handler with stopPropagation? The tooltip shouldn't be keyboard focusable anyways and tooltips aren't accessible on touch devices so no need for all the extra handling for those right?

**Comment by snowystinger** [comment]
I think it's fine to stop it with usePress. Because it's more than just pointerDown, we currently still support devices that don't have pointer events, so the full list would be
```
onPointerDown
onMouseDown
onClick
```

Might also need it for touch? and maybe the up/move/enter/leave? Seemed easier to let usePress just do it's thing, which is to block anything that might be perceived as a click.

**Comment by snowystinger** [comment]
Reopening because we'll solve this a different way
reverting - https://github.com/adobe/react-spectrum/pull/4535

> After a team discussion, we've decided to address this in all of our portalled components using something like https://github.com/facebook/react/issues/11387#issuecomment-1548827669
and provide guidance in our FAQ on how to ignore events that come through a portal.

**Pull request merged**